src/python/strelka/scanners/scan_transcode.py

```diff
@@ -0,0 +1,37 @@
+import io
+import logging
+
+import pillow_avif
+from PIL import Image
+from pillow_heif import register_heif_opener
+
+from strelka import strelka
+
+logging.getLogger("PIL").setLevel(logging.WARNING)
+
+# Must be imported as a plugin, doesn't need to be used
+_ = pillow_avif.AvifImagePlugin
+
+register_heif_opener()
+
+
+class ScanTranscode(strelka.Scanner):
+    """
+    Converts supported images to PNG for easier scanning
+
+    Typical supported output options:
+    gif webp jpeg bmp png tiff
+    """
+
+    def scan(self, data, file, options, expire_at):
+        output_format = options.get("output_format", "jpeg")
+
+        def convert(im):
+            with io.BytesIO() as f:
+                im.save(f, format=f"{output_format}", quality=90)
+                return f.getvalue()
+
+        # Send extracted file back to Strelka
+        self.emit_file(convert(Image.open(io.BytesIO(data))), name=file.name)
+
+        self.flags.append("transcoded")
```

Comment:
I know we discussed global exception handling - curious how you want to approach adding exceptions here. My first thought would be something like...

```
    def scan(self, data, file, options, expire_at):
        output_format = options.get("output_format", "jpeg")

        def convert(im):
            with io.BytesIO() as f:
                try:
                    im.save(f, format=f"{output_format}", quality=90)
                    return f.getvalue()
                except ValueError:
                  self.flags.append(f"{self.__class__.__name__} Exception:  Invalid format or quality.")
                except OSError:
                   self.flags.append(f"{self.__class__.__name__} Exception:  Unsupported format or invalid image file.")
                except AttributeError:
                   self.flags.append(f"{self.__class__.__name__} Exception:  Data is not a bytes-like object.")
                except Exception as e:
                   self.flags.append(f"{self.__class__.__name__} Exception: {str(e)[:50]}")

        # Send extracted file back to Strelka
        try:
            self.emit_file(convert(Image.open(io.BytesIO(data))), name=file.name)
        except Exception as e:
            self.flags.append(f"{self.__class__.__name__} Exception: Failed to emit file")
            return

        self.flags.append("transcoded")
```

Too much? Too specific to scanner?

Reply:
I catch `UnidentifiedImageError` now, which is what's thrown when a broken image is loaded in to Image.open. Fair to do if we expect a lot of exceptions from badly formatted or truncated image files. I'd like to add in other specific exceptions as they come up while running Strelka.

Added a test for a broken image.

I'm not keen on a broader catch. If emit_file() itself fails, I think that should be handled inside of emit_file(). I'd added code to add a flag when emit_file fails, and log the exception. If emit_file fails, it's likely due to a coordinator connectivity problem, and we shouldn't suppress those exceptions.